context.c

```diff
@@ -62,7 +65,15 @@ void ctx_free(struct Context **ptr)
   mutt_hash_free(&ctx->thread_hash);
   notify_free(&ctx->notify);
 
-  FREE(ptr);
+  STAILQ_FOREACH_SAFE(np, &contexts, entries, tmp)
+  {
+    if (np == ctx)
+    {
+      STAILQ_REMOVE(&contexts, np, Context, entries);
```

Comment:
`STAILQ_REMOVE` already does the looping for you.

Reply:
removed the loop in new commit 1c77574.